### ISSUE TITLE
Failure in GSI using ncio/1.1.0

### ISSUE THREAD

**Issue by MichaelLueken** [NOAA-EMC/NCEPLIBS-ncio#58]
While running the GSI using the ncio/1.1.0 stack module, the code fails with the following error message:

```
 calling general_read_gfsatm_nc
  in g_create_egrid2agrid, rlone(nlone) + dlone not within tolerance for 0 merid
 ian
  incorrect input grid coordinates in subroutine g_create_egrid2agrid, program s
 tops
application called MPI_Abort(MPI_COMM_WORLD, 999) - process 47
```

I added prints to `src/gsi/general_read_gfsatm.f90`. The following calls to `read_vardata`

```
       call read_vardata(atmges, 'grid_xt', rlons_tmp)
       call read_vardata(atmges, 'grid_yt', rlats_tmp)
```

return invalid values. All entries of `rlons_tmp` are `0.0`. All entries of `rlats_tmp` are `0.0` except the first which is `89.2842275325136`.

I was able to get around this failure by adding:

`ncerr = nf90_get_var(dset%ncid, dset%variables(nvar)%varid, values)`

back into read_vardata_code_1d.f90:

```
    if (present(ncstart) .and. present(nccount)) then
       allocate(values(nccount(1)))
       start(1)=ncstart(1); count(1)=nccount(1)
       if (dset%variables(nvar)%ndims == 2) then
         start(2)=1; count(2)=1
       end if
    else
       if (dset%variables(nvar)%ndims == 2) then
          allocate(values(dimlen))
       else
          allocate(values(dset%variables(nvar)%dimlens(1)))
          ncerr = nf90_get_var(dset%ncid, dset%variables(nvar)%varid, values)
       end if
    end if
    ncerr = nf90_get_var(dset%ncid, dset%variables(nvar)%varid, values,&
                         start=start, count=count)
```

**Comment by jswhit** [comment]
@MichaelLueken-NOAA I see the problem and will issue a PR for NCEPLIBS_ncio shortly.  The regression tests lacked coverage for 1d arrays, which I will also fix.

**Comment by jswhit** [comment]
PR #59 should fix this, but I haven't yet tested with GSI

**Comment by MichaelLueken** [comment]
@jswhit I tested your change in the GSI.  First, I applied your modification to read_vardata_code_1d.f90 to what is currently in ncio/1.1.0, rebuilt the ncio/1.1.0 module from the stack, and compiled Rahul's feature/cmake-update branch with this updated ncio/1.1.0 module.  The test successfully ran through to completion without issue.  The updates in PR #59 fixes the problem encountered in the GSI associated with ncio/1.1.0.
